hoomd/pytest/test_dcd.py

```diff
@@ -0,0 +1,66 @@
+import hoomd
+import pytest
+import numpy as np
+
+
+def test_attach(simulation_factory, two_particle_snapshot_factory, tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
+    filename = d / "temporary_test_file.dcd"
+    sim = simulation_factory(two_particle_snapshot_factory())
+    dcd_dump = hoomd.write.DCD(filename, hoomd.trigger.Periodic(1))
+    sim.operations.add(dcd_dump)
+    sim.operations._schedule()
+    for i in range(10):
+        sim.run(1)
+
+
+def test_set_period(tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
+    filename = d / "temporary_test_file.dcd"
+    dcd_dump = hoomd.write.DCD(filename, hoomd.trigger.Periodic(1))
+    with pytest.raises(RuntimeError):
+        dcd_dump.set_period(1)
+
+
+def test_enabled(tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
+    filename = d / "temporary_test_file.dcd"
+    dcd_dump = hoomd.write.DCD(filename, hoomd.trigger.Periodic(1))
+    dcd_dump.enabled = False
+    with pytest.raises(RuntimeError):
+        dcd_dump.enable()
+
+
+def test_write(simulation_factory, two_particle_snapshot_factory, tmp_path):
+    garnett = pytest.importorskip("garnett")
```

Comment:
Nice! I didn't know this API existed. This is very clean.

I will rebuild the CI images with garnett so this test can run.